generic_server.cc

```diff
@@ -156,9 +156,10 @@ server::listen(socket_address addr, std::shared_ptr<seastar::tls::credentials_bu
             }
         });
     }
-    return f.then([this, addr, is_shard_aware, keepalive](shared_ptr<seastar::tls::server_credentials> creds) {
+    return f.then([this, addr, is_shard_aware, keepalive, unix_domain_socket_permissions](shared_ptr<seastar::tls::server_credentials> creds) {
         listen_options lo;
         lo.reuse_address = true;
+        lo.unix_domain_socket_permissions = unix_domain_socket_permissions;
```

Comment:
question: Does this depend on modifying `seastar::net::listen_options` in Seastar?